xbmc/interfaces/json-rpc/InputOperations.cpp

```diff
@@ -84,6 +86,32 @@ JSONRPC_STATUS CInputOperations::ExecuteAction(const std::string &method, ITrans
   return SendAction(action);
 }
 
+JSONRPC_STATUS CInputOperations::ButtonEvent(const std::string& method,
+                                             ITransportLayer* transport,
+                                             IClient* client,
+                                             const CVariant& parameterObject,
+                                             CVariant& result)
+{
+  std::string button = parameterObject["button"].asString();
+  std::string keymap = parameterObject["keymap"].asString();
+  int holdtime = parameterObject["holdtime"].asInteger();
```

Comment:
To avoid adding a compiler warning on windows, can you do:

```c++
int holdtime = static_cast<int>(parameterObject["holdtime"].asInteger());
```

and then validate that holdtime isn't < 0 to avoid wraparound when assigning to `newEvent->keybutton.holdtime`